### DISCUSSION TITLE
The protocol is inconsistent about sending a response to a canceled request.

### DISCUSSION THREAD

**Discussion by PBundyra** [build-server-protocol/build-server-protocol#513] [General]
If we cancel Compile/Test/Run request, the protocol requires us to note that by using the statusCode member in the Compile/Test/Run Result structure, filling the result member in the response message. However, the specification of the LSP Protocol requires us to note that by using the error member in the response message structure, and prohibits us from filling the result member in the response message.

**Comment by agluszak** [comment]
We should handle cancellation the same way as LSP does: https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#errorCodes

If the request was cancelled by the server, it should reply with code `-32802`, if it was cancelled by the client - with code `-32800`
